controllers/tykoasapidefinition_controller.go

```diff
@@ -430,3 +443,35 @@ func (r *TykOasApiDefinitionReconciler) deleteOasTpl(
 
 	return r.Update(ctx, tykOas)
 }
+
+func (r *TykOasApiDefinitionReconciler) reconcileOasTpl(ctx context.Context, tpl *v1alpha1.TykOasApiDefinition) error {
+	ingressList := networkingv1.IngressList{}
+
+	err := r.List(ctx, &ingressList, client.InNamespace(tpl.Namespace))
+	if err != nil {
+		return client.IgnoreNotFound(err)
+	}
+
+	for i := range ingressList.Items {
+		if ingressList.Items[i].GetAnnotations()[keys.IngressTemplateAnnotation] == tpl.Name {
+			key := client.ObjectKey{
+				Namespace: ingressList.Items[i].GetNamespace(),
+				Name:      ingressList.Items[i].GetName(),
+			}
+
+			r.Log.Info("Updating ingress " + key.String())
+
+			if ingressList.Items[i].Labels == nil {
+				ingressList.Items[i].Labels = make(map[string]string)
+			}
+
+			ingressList.Items[i].Labels[keys.IngressTaintLabel] = strconv.FormatInt(time.Now().UnixNano(), 10)
```

Comment:
it triggers an update on ingress object which helps to update existing oas templates or remove orphan apis.

Reply:
I got confused because of [Taints](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/)
Have you intentionally set the same name?